static/app/components/profiling/flamegraphZoomView.tsx

```diff
@@ -782,4 +802,112 @@ const Canvas = styled('canvas')`
   position: absolute;
 `;
 
+function selectNearestFrame(n: FlamegraphFrame, dir: 'up' | 'down' | 'left' | 'right') {
+  if (!n) {
+    return null;
+  }
+
+  const targetDepth = n.depth;
+  let parent = n.parent;
+  let node = n;
+
+  if (dir === 'up' && parent) {
+    return parent;
+  }
+
+  const child = n.children?.[0];
+  if (dir === 'down' && child) {
+    return child;
+  }
+
+  while (parent) {
+    const indexOfChild = parent.children.indexOf(node);
+    if (indexOfChild === -1) {
+      return n;
+    }
+    const hasSiblings =
+      dir === 'right' ? indexOfChild < parent.children.length - 1 : indexOfChild > 0;
+
+    if (hasSiblings) {
+      const siblingOffset = dir === 'right' ? 1 : -1;
+      const sibling = parent.children[indexOfChild + siblingOffset];
+      const foundNode = scanForNearestFrameWithDepth(
+        sibling,
+        targetDepth,
+        dir as 'left' | 'right'
+      );
+      return foundNode;
+    }
+
+    node = parent;
+    parent = parent.parent;
+  }
+  return null;
+}
+
+function scanForNearestFrameWithDepth(
+  n: FlamegraphFrame,
+  depth: number,
+  dir: 'left' | 'right'
+) {
+  const stack = [n];
+
+  while (stack.length) {
+    const node = stack.pop();
+    if (!node) {
+      return node;
+    }
+    if (node.depth === depth) {
+      return node;
+    }
+    if (!node.children) {
+      continue;
+    }
+
+    const nextNode = node.children[dir === 'right' ? 0 : node.children.length - 1];
+    if (!nextNode) {
+      return node;
+    }
+    stack.push(nextNode);
+  }
+
+  return n;
+}
+
+const keyDirectionMap = {
+  ArrowUp: 'up',
+  ArrowDown: 'down',
+  ArrowLeft: 'left',
+  ArrowRight: 'right',
+  Tab: 'right',
+  'Shift+Tab': 'left',
+  Enter: 'down',
+  'Shift+Enter': 'up',
+  w: 'up',
+  s: 'down',
+  a: 'left',
+  d: 'right',
+};
```

Comment:
stolen from my days as a spreadsheet wizard.

Reply:
I think it's ok, the only one I'm not sure is enter and shift+enter. I'd probably remove those for now and in the future use this to zoom to frame

Reply:
i'd drop the enter keybind and make that the default behavior of tab. since we implemented cycling with up/down, we can just make tab cycle by moving down, and shift+tab moving up